### PR TITLE
Fix missing GMPO loss terms

### LINKED ISSUE #6808: GMPO: use_liger_kernel=True silently bypasses the GMPO loss; MoE aux loss and entropy bonus dropped (gaps not covered by #6095)
### Description

Follow-up to #6056 and complementary to open PR #6095 (which addresses the vLLM IS correction, off-policy masking, the zero-advantage clip branch, and the `loss_type` default). Two remaining gaps in `trl/experimental/gmpo/gmpo_trainer.py` are **not** covered by that PR (at f72460224c5477e65031b2363f54368699105182):

### 1. `use_liger_kernel=True` silently bypasses the GMPO loss entirely

`GMPOTrainer` only overrides `_compute_loss`, but the inherited `compute_loss` routes around it when liger is enabled:

```python
# trl/trainer/grpo_trainer.py (inherited by GMPOTrainer)
def compute_loss(self, model, inputs, return_outputs=False, num_items_in_batch=None):
    ...
    if self.use_liger_kernel:
        # Compute the loss using the liger grpo loss
        unwrapped_model = self.accelerator.unwrap_model(model)
        return self._forward_redirection(model, unwrapped_model, self.compute_liger_loss, unwrapped_model, inputs)
    return self._compute_loss(model, inputs)
```

The init-time guard in main GRPO only rejects liger for `importance_sampling_level` other than `token`/`sequence`, which GMPO doesn't change — so `GMPOTrainer` with `use_liger_kernel=True` silently trains with the plain GRPO clipped-surrogate loss instead of GMPO's geometric-mean objective. (`gspo_token` is incidentally protected by that guard via `importance_sampling_level="sequence_token"`; GMPO is not.) The trainer should either raise on `use_liger_kernel=True` or route the liger path to a GMPO implementation.

### 2. MoE aux loss and entropy bonus silently dropped

GMPO's `_compute_loss` calls `_get_per_token_logps_and_entropies` without `compute_aux_loss=self.aux_loss_enabled` and never adds the `router_aux_loss_coef * aux_loss` term, so MoE models lose the load-balancing loss. The `_entropy_bonus_enabled` block from main GRPO is also absent, so `entropy_coef`/adaptive-entropy settings are silently ignored. (Minor, same family: the forward kwargs are missing `spatial_shapes`/`num_tiles` for LFM2-VL-style models.)

Per the repo's consistency policy for duplicated trainer blocks, these should be re-aligned with `trl/trainer/grpo_trainer.py::_compute_loss`, keeping the geometric-mean objective as the intentional divergence.

I'm happy to submit a PR (rebasing on #6095 if it lands first).

### Reproduction

Code inspection; for item 1, instantiate `GMPOTrainer` with `use_liger_kernel=True` and observe that `GMPOTrainer._compute_loss` is never called (the liger GRPO loss is used instead).

### System Info

Found by code inspection (AI-assisted audit) at f72460224c5477e65031b2363f54368699105182.

- Platform: macOS-26.5.1-arm64-arm-64bit
- Python version: 3.12.7
- TRL version: 1.5.1+f724602
- PyTorch version: 2.7.1
- Transformers version: 5.8.1
- Accelerate version: 1.8.1

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by guptaishaan**
# What does this PR do?

Fixes #6808

`GMPOTrainer` inherited the Liger dispatch to the fused GRPO objective, bypassing GMPO's geometric-mean loss. This change rejects the unsupported combination and restores the MoE auxiliary loss, entropy bonus, and missing multimodal forward arguments in GMPO's duplicated loss code.

The focused regression test failed before the fix and passes after it. All 12 GMPO tests and the configured pre-commit checks pass on Linux using CPU execution. GPU execution and a real MoE training run were not verified because the resolved PyTorch CUDA build requires a newer driver than the node provides.

Thanks to @mmjerge for reporting the issue and identifying the affected paths.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

No documentation update is needed for this bug fix.

## AI writing disclosure

We welcome the use of AI tools to help with contributions. For transparency and to help us improve our review process, please indicate the level of AI involvement in this PR.

- [ ] No AI usage: the PR was written entirely by a human.
- [ ] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.
- [x] AI-generated: the PR was mostly or fully generated by an AI tool.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR.

---

> [!NOTE]
> **Medium Risk**
> Changes core RL training loss composition for GMPO (entropy controller state, aux loss scaling); behavior should match GRPO but affects MoE/multimodal GMPO runs.
> 
> **Overview**
> **`GMPOTrainer`** overrides **`_compute_loss`** for the geometric-mean objective, but that path was missing pieces that **`GRPOTrainer`** already applies: MoE router auxiliary loss, entropy regularization (static and adaptive), and multimodal forward kwargs **`spatial_shapes`** / **`num_tiles`**.
> 
> This PR wires those terms back into GMPO’s loss (including gradient-accumulation scaling and metrics for **`policy_loss`**, **`aux_loss`**, and **`entropy_coef`**), and **raises `NotImplementedError`** when **`use_liger_kernel=True`** so training cannot silently use the fused GRPO path instead of GMPO’s loss.
> 
> Regression coverage adds a unit test for the combined aux + entropy loss math and an init test for the Liger rejection.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b7bec1f739d187803fcd07107c58a4fc1128a072. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by qgallouedec** [comment]
Closing: we don't review fully AI-generated PRs from first-time contributors, see the [AI usage policy](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#ai-usage-policy).
